src/Metadata/Util/PropertyInfoToTypeInfoHelper.php

```diff
@@ -0,0 +1,150 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Metadata\Util;
+
+use Symfony\Component\PropertyInfo\Type as LegacyType;
+use Symfony\Component\TypeInfo\Exception\InvalidArgumentException;
+use Symfony\Component\TypeInfo\Type;
+use Symfony\Component\TypeInfo\Type\BuiltinType;
+use Symfony\Component\TypeInfo\Type\UnionType;
+use Symfony\Component\TypeInfo\TypeIdentifier;
+
+/**
+ * A helper about PropertyInfo Type conversion.
+ *
+ * @see https://github.com/mtarld/symfony/commits/backup/chore/deprecate-property-info-type/
+ *
+ * @author Mathias Arlaud <mathias.arlaud@gmail.com>
+ *
+ * @internal
+ */
+final class PropertyInfoToTypeInfoHelper
```

Comment:
I think this is good enough for now, but in the future we should migrate to use the new type class by default and convert it to the legacy type instead of the other way. This will likely improve performance, and will ease the maintenance.

Reply:
yes but while its experimental this is a good solution